controllers/storagecluster/kms_resources.go

```diff
@@ -39,6 +41,50 @@ var (
 	_ kmsConfigMapValidateFunc = reachKMSProvider
 )
 
+func deleteKMSResoucres(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	// if 'KMS' is not enabled, nothing to delete
+	if !sc.Spec.Encryption.KeyManagementService.Enable {
+		return nil
+	}
+	var allGetErrors []error = make([]error, 2)
+	var allRuntimeObjs []runtime.Object = make([]runtime.Object, 2)
+	allRuntimeObjs[0], allGetErrors[0] = getKMSConfigMap(sc, r.Client, nil)
+	allRuntimeObjs[1], allGetErrors[1] = getKMSSecretToken(sc, r.Client)
+	allKMSResourceNames := []string{KMSConfigMapName, KMSTokenSecretName}
```

Comment:
Instead of doing all this, it would probably be more idiomatic to use goroutines, channels, and WaitGroups, with an inline struct something like this:

```
kmsResources := []{
  Name string
  Get KMSGetFunc
}{
  {KMSConfigMapName, getKMSConfigMap},
  {KMSTokenSecretName, getKMSSecretToken},
}
```

My syntax may be off. Yes this would mean refactoring `getKMSConfigMap()` to not take a validation function, but I'd want that to happen anyway. Since it only gets called twice and only one of those times actually uses a validation function I think it's a reasonable refactor.

Reply:
I have implemented with the loop logic itself (not changed to the `go` functions and channels yet)... Please take a look and let me know...